### PR TITLE
Feat: Run linters against staged git files

### PR DESCRIPTION
**Pull request by DamianArado**
Closes: #578 

Checked it locally, seems to work fine. You can check on your system as well if you're using MacOS or Linux. I'm using Windows.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "eslint": "^8.39.0",
     "eslint-config-prettier": "^8.8.0",
     "eslint-plugin-prettier": "^4.2.1",
+    "husky": "^8.0.3",
```
[the remark this answers is not in the document]
Comment (DamianArado) [reply package.json:34]: We don't need to install it since we do `npx lint-staged`.

**Comment by 9inpachi** [comment]
Ahan, I see. I think we should make it a dev dependency instead. Since it's
supposed to be there with husky.

**Comment by DamianArado** [comment]
I followed this:
https://prettier.io/docs/en/precommit.html#option-1-lint-stagedhttpsgithubcomokonetlint-staged

Will this be effective in our use case? If not, I'll add it as a dev
dependency and perform those checks again.

**Comment by 9inpachi** [comment]
We are essentially using lint-staged as an independent CLI tool which can
lead to inconsistent developer experience on different machines. If we
install it as a dev dependency and call the npm script for it, we can make
the setup consistent for everyone. This will also ensure that everyone is
using the same version (working) version of lint-staged.

**Comment by DamianArado** [comment]
Aye, aye, captain!

**Comment by DamianArado** [comment]
I looked into it and I think the only way to run lint-staged after installing it as a dependency is to
do `node ./node_modules/lint-staged/bin/lint-staged`

but this doesn't work as expected
```
$ git commit -m "test this setup"
[STARTED] Preparing lint-staged...
[SUCCESS] Preparing lint-staged...
[STARTED] Running tasks for staged files...
[STARTED] .lintstagedrc.json — 6 files
[STARTED] *.ts — 2 files
[STARTED] *.{js,html,scss,json} — 2 files
[STARTED] eslint --fix
[STARTED] prettier --write
[FAILED] eslint --fix [FAILED]
[FAILED] eslint --fix [FAILED]
[FAILED] eslint --fix [FAILED]
[STARTED] Applying modifications from tasks...
[SKIPPED] Skipped because of errors from tasks.
[STARTED] Reverting to original state because of errors...
[FAILED] prettier --write [FAILED]
[FAILED] prettier --write [FAILED]
[SUCCESS] Reverting to original state because of errors...
[STARTED] Cleaning up temporary files...
[SUCCESS] Cleaning up temporary files...

✖ eslint --fix:
node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module '../package.json'
Require stack:
- D:\phoenix\node_modules\eslint\bin\eslint.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at process.onFatalError (D:\phoenix\node_modules\eslint\bin\eslint.js:103:25)
    at process.emit (node:events:513:28)
    at process._fatalException (node:internal/process/execution:149:25) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ 'D:\\phoenix\\node_modules\\eslint\\bin\\eslint.js' ]
}

Node.js v18.16.0

✖ prettier --write:
node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module './package.json'
Require stack:
- D:\phoenix\node_modules\prettier\bin-prettier.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (D:\phoenix\node_modules\prettier\bin-prettier.js:61:19)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ 'D:\\phoenix\\node_modules\\prettier\\bin-prettier.js' ]
}

Node.js v18.16.0
husky - pre-commit hook exited with code 1 (error)
```

Maybe that's why they suggest using npx [here](https://github.com/okonet/lint-staged#command-line-flags) or am I missing a trick?

**Comment by 9inpachi** [comment]
You can use `yarn` to call the executable from `./node_modules/lint-staged/bin` through `yarn lint-staged`. I pushed a commit that should work. I was testing it out to confirm and it seems to work so I just pushed the change.

**Review by 9inpachi — APPROVED** [review]

**Comment by DamianArado** [comment]
@9inpachi gracias!

**Pull request merged**